common-npm-packages/docker-common-v2/containerimageutils.ts

```diff
@@ -72,4 +73,125 @@ export function getResourceName(image: string, digest: string) {
     namespace = namespace ? namespace + '/' : '';
     
     return "https://" + registry  + namespace  + repository + "@sha256:" + digest;
-  }
\ No newline at end of file
+  }
+
+export function getFinalBaseImageName(dockerFilePath: string): string {
+    // This method takes into consideration multi-stage dockerfiles, it tries to find the final
+    // base image for the container.
+    // ex:
+    // FROM ubuntu:16.04 as builder
+    //
+    // FROM builder as base
+    // RUN echo 'test'
+    //
+    // FROM base
+    // RUN echo 'test2'
+    // 
+    // This code is going to return ubuntu:16.04
+
+    try {
+        const dockerFileContent = fs.readFileSync(dockerFilePath, 'utf-8');
+
+        if (!dockerFileContent || dockerFileContent == "") {
+            return "";
+        }
+
+        var lines = dockerFileContent.split(/[\r?\n]/);
+        var tagToImageNameMapping: Map<string, string> = new Map<string, string>();
+        var baseImage = "";
+        for (var i = 0; i < lines.length; i++) {
+            var index = lines[i].toUpperCase().indexOf("FROM");
+            if (index == -1) {
+                continue;
+            }
+
+            var nameComponents = lines[i].substring(index + 4).toLowerCase().split(" as ");
+            var prospectImageName = nameComponents[0].trim();
+            if (nameComponents.length > 1) {
+                var tag = nameComponents[1].trim();
```

Comment:
Use names `alias` and `aliasToImageNameMapping` as `tag` has a different meaning in Docker context.